File: Taller Tipos de clases/C#/Clase Interfaz/Vehiculo.cs
interface IVehiculo
{
    void Encender();
}

class Moto : IVehiculo
{
    public string Modelo;

    public Moto()
    {
        Modelo = "NA";
    }

    public Moto(string modelo)
    {
        Modelo = modelo;
    }

    public Moto(Moto m)
    {
        Modelo = m.Modelo;
    }

    public void Encender() { }
}

File: Taller Tipos de clases/Go/Clase Interfaz/Vehiculo.go
interface Vehiculo {
    void encender();
}

class Moto implements Vehiculo {
    String modelo;

    public Moto() {
        modelo = "NA";
    }

    public Moto(String modelo) {
        this.modelo = modelo;
    }

    public Moto(Moto m) {
        this.modelo = m.modelo;
    }

    @Override
    public void encender() {}
}

